inc/Engine/Common/PerformanceHints/AbstractFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+declare( strict_types=1 );
+
+namespace WP_Rocket\Engine\Common\PerformanceHints;
+
+interface AbstractFactory {
```

Comment:
I would suggest to apply a different naming convention here, as when I read `AbstractFactory` I expect an abstract class, not an interface.

`FactoryInterface` would better match with what this class is